src/logics/ArithLogic.h

```diff
@@ -160,12 +162,20 @@ class ArithLogic : public Logic {
     bool isIntNeg(SymRef sr) const { return sr == sym_Int_NEG; }
     bool isRealNeg(SymRef sr) const { return sr == sym_Real_NEG; }
 
-    bool isTimes(SymRef sr) const { return isIntTimes(sr) or isRealTimes(sr); }
+    bool isTimes(SymRef sr) const { return isTimesLin(sr) or isTimesNonlin(sr); };
+    bool isTimesLin(SymRef sr) const { return isIntTimes(sr) or isRealTimes(sr); }
+    bool isTimesNonlin(SymRef sr) const { return isIntTimesNonlin(sr) or isRealTimesNonlin(sr); }
     bool isTimes(PTRef tr) const { return isTimes(getPterm(tr).symb()); }
+    bool isTimesLin(PTRef tr) const { return isTimesLin(getPterm(tr).symb()); }
+    bool isTimesNonlin(PTRef tr) const { return isTimesNonlin(getPterm(tr).symb()); }
     bool isIntTimes(PTRef tr) const { return isIntTimes(getPterm(tr).symb()); }
+    bool isIntTimesNonlin(PTRef tr) const { return isIntTimesNonlin(getPterm(tr).symb()); }
     bool isRealTimes(PTRef tr) const { return isRealTimes(getPterm(tr).symb()); }
+    bool isRealTimesNonlin(PTRef tr) const { return isRealTimesNonlin(getPterm(tr).symb()); }
     bool isIntTimes(SymRef sr) const { return sr == sym_Int_TIMES; }
```

Comment:
```suggestion
    bool isIntTimesLin(SymRef sr) const { return sr == sym_Int_TIMES_LIN; }
```
I think that the symbol `sym_Int_TIMES` no longer makes sense since it is internal. Predicates like `isTimes` can stay e.g. in the form
```
    bool isTimes(SymRef sr) const { return isTimesLin(sr) or isTimesNonlin(sr); }
```

Reply:
I think `sym_Int_TIMES` is external imho, only `sym_Int_TIMES_NONLIN` is internal...
And if we make `sym_Int_TIMES_LIN` we will have two internal multiplications

Reply:
Changed